src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -3,102 +3,203 @@
+    EnrollmentRepository enrollmentRepository;
 
-    // instructor lists assignments for a section.  Assignments ordered by due date.
+    @Autowired
+    SectionRepository sectionRepository;
+
+    // instructor lists assignments for a section. Assignments ordered by due date.
     // logged in user must be the instructor for the section
     @GetMapping("/sections/{secNo}/assignments")
     public List<AssignmentDTO> getAssignments(
             @PathVariable("secNo") int secNo) {
 
-        // TODO remove the following line when done
-		
-		// hint: use the assignment repository method 
-		//  findBySectionNoOrderByDueDate to return 
-		//  a list of assignments
+        // hint: use the assignment repository method
+        // findBySectionNoOrderByDueDate to return
+        // a list of assignments
+
+        // Fetch assignments from repository
+        List<Assignment> assignments = assignmentRepository.findBySectionNoOrderByDueDate(secNo);
+
+        if (assignments.size() == 0) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Section " + secNo + " not found");
+        }
 
-        return null;
+        // Convert Assignment entities to DTOs
+        return assignments.stream()
+            .map(assignment -> new AssignmentDTO(
+                assignment.getAssignmentId(),
+                assignment.getTitle(),
+                assignment.getDueDate().toString(),
+                assignment.getSection().getCourse().getCourseId(),
+                assignment.getSection().getSecId(),
+                assignment.getSection().getSectionNo()))
+            .collect(Collectors.toList());
     }
 
     // add assignment
     // user must be instructor of the section
     // return AssignmentDTO with assignmentID generated by database
     @PostMapping("/assignments")
     public AssignmentDTO createAssignment(
-            @RequestBody AssignmentDTO dto) {
+            @RequestBody AssignmentDTO assignmentDTO) {
 
-        // TODO remove the following line when done
+        Section s = sectionRepository.findById(assignmentDTO.secId()).orElse(null);
+        if (s == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Section " + s.getSecId() + " not found");
+        }
 
-        return null;
+        Assignment a = new Assignment();
+        a.setSection(s);
+
+        // assignmentId generated by database
+
+        return getAssignmentDTO(assignmentDTO, a);
     }
 
-    // update assignment for a section.  Only title and dueDate may be changed.
+    // update assignment for a section. Only title and dueDate may be changed.
     // user must be instructor of the section
     // return updated AssignmentDTO
     @PutMapping("/assignments")
-    public AssignmentDTO updateAssignment(@RequestBody AssignmentDTO dto) {
+    public AssignmentDTO updateAssignment(@RequestBody AssignmentDTO assignmentDTO) {
+
+        Assignment a = assignmentRepository.findById(assignmentDTO.id()).orElse(null);
```

Comment:
Alternative: `public AssignmentDTO updateAssignment(@RequestBody AssignmentDTO assignmentDTO) {
Assignment a = assignmentRepository.findById(assignmentDTO.id()).orElseThrow(() ->
new ResponseStatusException(HttpStatus.NOT_FOUND, "Assignment not found for ID: " + assignmentDTO.id()));
return getAssignmentDTO(assignmentDTO, a);
    }`

Reply:
Suggestion implemented.